internal/command/postgres/config_update.go

```diff
@@ -164,70 +179,69 @@ func runMachineConfigUpdate(ctx context.Context, app *api.AppCompact) error {
-	)
-
-	var MinPostgresVersion = "v0.0.32"
+func updateStolonConfig(ctx context.Context, app *api.AppCompact, leaderIP string) (bool, error) {
+	io := iostreams.FromContext(ctx)
 
-	if err := hasRequiredVersionOnNomad(app, MinPostgresVersion, MinPostgresVersion); err != nil {
-		return err
+	restartRequired, changes, err := resolveConfigChanges(ctx, app, flypg.StolonManager, leaderIP)
+	if err != nil {
+		return false, err
 	}
 
-	client := client.FromContext(ctx).API()
-
-	agentclient, err := agent.Establish(ctx, client)
+	fmt.Fprintln(io.Out, "Performing update...")
+	cmd, err := flypg.NewCommand(ctx, app)
 	if err != nil {
-		return errors.Wrap(err, "can't establish agent")
+		return false, err
 	}
 
-	pgInstances, err := agentclient.Instances(ctx, app.Organization.Slug, app.Name)
+	err = cmd.UpdateSettings(ctx, leaderIP, changes)
 	if err != nil {
-		return fmt.Errorf("failed to lookup 6pn ip for %s app: %v", app.Name, err)
-	}
-	if len(pgInstances.Addresses) == 0 {
-		return fmt.Errorf("no 6pn ips found for %s app", app.Name)
+		return false, err
 	}
+	fmt.Fprintln(io.Out, "Update complete!")
 
-	leaderIP, err := leaderIpFromNomadInstances(ctx, pgInstances.Addresses)
+	return restartRequired, nil
+}
+
+func updateFlexConfig(ctx context.Context, app *api.AppCompact, leaderIP string) (bool, error) {
+	var (
+		io     = iostreams.FromContext(ctx)
+		dialer = agent.DialerFromContext(ctx)
+	)
+
+	restartRequired, changes, err := resolveConfigChanges(ctx, app, flypg.ReplicationManager, leaderIP)
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	requiresRestart, err := updateStolonConfig(ctx, app, leaderIP)
+	fmt.Fprintln(io.Out, "Performing update...")
+	leaderClient := flypg.NewFromInstance(leaderIP, dialer)
+
+	// Push configuration settings to consul.
+	if err := leaderClient.UpdateSettings(ctx, changes); err != nil {
+		return false, err
+	}
+
+	machines, err := mach.ListActive(ctx)
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	if requiresRestart {
-		if !autoConfirm {
-			fmt.Fprintln(io.Out, colorize.Yellow("Please note that some of your changes will require a cluster restart before they will be applied."))
+	// Sync configuration settings for each node. This should be safe to apply out-of-order.
+	for _, machine := range machines {
+		client := flypg.NewFromInstance(machine.PrivateIP, dialer)
 
-			switch confirmed, err := prompt.Confirm(ctx, "Restart cluster now?"); {
-			case err == nil:
-				if !confirmed {
-					return nil
-				}
-			case prompt.IsNonInteractive(err):
-				return prompt.NonInteractiveError("yes flag must be specified when not running interactively")
-			default:
-				return err
-			}
-		}
-
-		if err := nomadRestart(ctx, app); err != nil {
-			return err
+		// Pull configuration settings down from Consul for each node and reload the config.
+		err := client.SyncSettings(ctx)
+		if err != nil {
+			return false, fmt.Errorf("failed to sync configuration on %s: %s", machine.ID, err)
```

Comment:
We may want to introduce a `fly pg config reload/sync` command so this can be re-run.  It would be a bummer to have the user do a full restart to apply run-time settings.

Reply:
I created a ticket for this and will swing back to it.